### PR TITLE
Teuchos: added StackedTimer option to print a timer only if time > dr…

### LINKED ISSUE #6289: Teuchos: Have stacked time summary have a minimum report time
## Enhancement
@trilinos/tempus 

I'm seeing a ton of 0 cost things in the timing log file, I'd like to have a minimum value for run time to count

```
|   |   runEvaluators: 362.898 - 17.9087% [6000] {min=198.397, max=629.832, std dev=140.738} <12, 19, 8, 3, 0, 5, 0, 0, 13, 4>
|   |   |   BC1 - AfterSolve: 0.00091644 - 0.000252534% [1000] {min=0.00077158, max=0.00107357, std dev=7.58366e-05} <3, 8, 5,\

```

SO, this is about 1e-6 and could be ignored.  So adding an arg to the dump command

### PR DESCRIPTION
**Pull request by rppawlo**
Closes #6289

### CODE REVIEW

File: packages/teuchos/comm/src/Teuchos_StackedTimer.cpp [hunk 1]
```diff
@@ -192,6 +192,8 @@ StackedTimer::computeColumnWidthsForAligment(std::string prefix,
   double total_time = 0.0;
 
   for (int i=0; i<flat_names_.size(); ++i ) {
+    if (sum_[i]/active_[i] < options.drop_time)
```
Comment (mhoemmen) [inline packages/teuchos/comm/src/Teuchos_StackedTimer.cpp]: Wouldn't you want `<= 0`, if the drop time is zero?

Comment (rppawlo) [reply packages/teuchos/comm/src/Teuchos_StackedTimer.cpp]: I wanted to make sure that all timers print if drop tolerance is zero. It should never happen in practice but... Happy to change this if you want. I could then make the default -1.0 to force all timers to print.

Comment (rppawlo) [reply packages/teuchos/comm/src/Teuchos_StackedTimer.cpp]: fixed.

**Comment by rppawlo** [comment]
@mhoemmen could you review again?

File: packages/teuchos/comm/src/Teuchos_StackedTimer.cpp (hunk 1 above)
Comment (mhoemmen) [reply packages/teuchos/comm/src/Teuchos_StackedTimer.cpp]: @rppawlo Thanks! :-D

**Pull request merged**